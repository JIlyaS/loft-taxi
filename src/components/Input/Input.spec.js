import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import {Input} from './';

const defaultProps = {
  type: 'text',
  name: '',
  label: '',
  placeholder: '',
  value: '',
  classNameWrap: '',
  isAutofocus: false,
  isRequired: false,
  onChange: () => {},
};

const TestInput = props => <Input {...defaultProps} {...props} />;

describe('Input', () => {
  it('renders correctly snapshot', () => {
      const tree = renderer.create(<TestInput />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it("render card correctly with null props", () => {
      const props = {
        type: 'text',
        name: '',
        label: '',
        placeholder: '',
        value: '',
        classNameWrap: '',
        isAutofocus: false,
        isRequired: false,
        onChange: () => {},
      };
      const InputComponent = mount(<TestInput {...props} />);
      expect(InputComponent.prop('type')).toEqual('text');
      expect(InputComponent.prop('value')).toEqual('');
    });
});
