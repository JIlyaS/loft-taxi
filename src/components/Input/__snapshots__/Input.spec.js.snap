// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly snapshot 1`] = `
<div
  className="input"
>
  <div
    className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
        disabled={false}
        id=""
        name=""
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
